### ISSUE TITLE
opt: group by on query causes query to run very slowly

### ISSUE THREAD

**Issue by roncrdb** [cockroachdb/cockroach#43101]
**Describe the problem**

The following query run very slowly: 
```
SELECT max(r.reported) AS lastrun, IF(max(r.reported) IS NULL, false, true) AS haveresult FROM results@results_query_reported AS r WHERE r.query = ANY (ARRAY['TzV1Gyi9I5iv59IRrh9_fw', '123']);

SELECT r.query, max(r.reported) AS lastrun, IF(max(r.reported) IS NULL, false, true) AS haveresult FROM results AS r WHERE r.query = ANY (ARRAY['TzV1Gyi9I5iv59IRrh9_fw']) group by r.Query;
```

Attached are the explain (opt env) and stats for both queries: 

[Explains}(https://docs.google.com/document/d/1-7vHk53N_qJtMBuWdPjvW_DazyKqOwsM9pNWq_q_b5U/edit?usp=sharing)

[Statistics](https://drive.google.com/file/d/1yNUyuRcUC3nKnQCVoUNtnsGcmRWerzhz/view?usp=sharing)

**Expected behavior**
Could the queries be optimized in such a way that they return quicker. 

**Environment:**
 - CockroachDB version 19.2

**Comment by RoachietheSupportRoach** [comment]
Zendesk ticket #[4229](https://cockroachdb.zendesk.com/agent/tickets/4229) has been linked to this issue.

**Comment by RaduBerinde** [comment]
Posted more info in the ticket. The "fast" query uses the max -> reverse single row scan optimization, which doesn't apply when we're working with multiple IDs or grouping. Addressing this would require more complicated optimizer rules and special execution support (to scan at most 1 row inside each span).

**Comment by roncrdb** [comment]
@RaduBerinde, 

They reported back saying that the rewrite was slower than the original. 

` the average/random set of ids in the array is much slower, around 150ms`

I have the explain analyze and opt env if you need them, let me know.

**Comment by RaduBerinde** [comment]
I looked in the ticket. I was under the impression that we have a small number of IDs. The rewritten query I suggested boils down to getting the result of each ID (sequentially). The slow query has quite a lot of IDs. I don't think there is much we can do about this case from the optimizer side. Even if we supported "skip scans" to just get a row for each ID, the "skipping" would still happen sequentially.
